### PR TITLE
updated sample to use latest azblob SDK package

### PR DESCRIPTION
**Pull request by Duffney**
## Purpose

* ...

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[x] Other... Please describe: Updated to use the latest SDK package.
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code

```
```

## What to Check
Verify that the following are valid
* ...

### CODE REVIEW

File: README.md
Comment (jhendrixMSFT) [inline README.md:51]: Since the sample is using AAD for authentication, you can remove these env vars.

File: storage-quickstart.go
Comment (jhendrixMSFT) [inline storage-quickstart.go:72]: It's typical for callers to use a struct literal instead of defining `blockOptions` separately.  So remove the declaration on line 64 and do this.
```go
_, err = blobClient.UploadBufferToBlockBlob(ctx, data, azblob.HighLevelUploadToBlockBlobOption{})
```

File: README.md
Comment (RickWinter) [inline README.md:19]: Nit: What feature requires 1.17?  Could we get away with saying 1.16 which is the current baseline for the SDK?  

What is the plan for these samples when we move to generics and require 1.18?

File: storage-quickstart.go
Comment (RickWinter) [inline storage-quickstart.go:36]: Should we do something like this which is guaranteed to not exist?
```suggestion
	url := "https://TODO_REPLACE_WITH_STORAGEACCOUNTNAME.blob.core.windows.net/" //replace with <StorageAccountName>
```

**Comment by RickWinter** [comment]
Josh can you add @zezha-msft from the storage team as a reviewer.

File: storage-quickstart.go
Comment (Duffney) [reply storage-quickstart.go:74]: So that's how you do that, thanks!

File: README.md
Comment (Duffney) [reply README.md:19]: Luckily, there are only a few existing service articles. When the change to 1.18 happens, let me know and I can get the versions updated. :) I'll start working on an include for that section so I can made one change that updates all docs.

**Comment by Duffney** [comment]
Once this PR gets merged, I have the article updated in a forked branch ready to be published.

**Review by jhendrixMSFT — APPROVED** [review]
Looks good to me but would be nice to get an ack from somebody on the storage team.

**Comment by Duffney** [comment]
@zezha-msft please review the changes and let me know if there's anything you'd like me to change. Or if I should add someone else as a reviewer. :) Thanks!

**Comment by Duffney** [comment]
If there are no objections, I'll merge this change on 12/29/2021.

**Review by zezha-msft — APPROVED** [review]
lgtm

**Comment by zezha-msft** [comment]
Sorry for the delayed reply. Thanks @Duffney for the sample!

**Comment by Duffney** [comment]
> Sorry for the delayed reply. Thanks @Duffney for the sample!

Not a problem @zezha-msft,  I appreciate you taking a look for me. :)

**Pull request merged**
